src/Compilers/VisualBasic/Test/Syntax/Scanner/ScannerTests.vb

```diff
@@ -504,13 +504,13 @@ Public Class ScannerTests
         Assert.Equal(" _", tk.ToFullString())
 
         tk = ScanOnce(" _ ", startStatement:=True)
-        Assert.Equal(SyntaxKind.EndOfFileToken, tk.Kind)
+        Assert.Equal(SyntaxKind.BadToken, tk.Kind)
```

Comment:
>SyntaxKind.BadToken [](start = 21, length = 19)

Did we change success case into an error case here?

Reply:
This case does not got through my new code I just return false and it goes to a higher level to deal with the same as before. If you think it should be an EndOfFileToken vs. SyntaxKind.BadToken I can look at changing it. This is the case I was dealing with the extra Else but I may have to deal with it here as well.

Reply:
> This case does not got through my new code I just return false and it goes to a higher level to deal with the same as before. If you think it should be an EndOfFileToken vs. SyntaxKind.BadToken I can look at changing it. This is the case I was dealing with the extra Else but I may have to deal with it here as well.

Even if that is due to just returning false, the behavior of the scanner has changed in this scenario. Should the behavior change and why?

---
In reply to: [213446793](https://github.com/dotnet/roslyn/pull/28327#discussion_r213446793) [](ancestors = 213446793)

Reply:
After more research It shouldn't, you are correct and I need to fix.

Reply:
Fixing https://github.com/dotnet/roslyn/pull/28327#pullrequestreview-150242464 fixes this. Thanks great catch.